### PR TITLE
(WIP) Add $vuetify.goTo() to smooth scroll to the given location

### PR DESCRIPTION
**Pull request by zaken**
## Description
Easily smooth scroll to a given page offset / anchor tag / element.

## Markup:
`<v-btn @click="$vuetify.goTo(123)">scroll</v-btn>`

## TODO:
- [ ] `<v-btn @click="$vuetify.goTo('#my-anchor')">scroll</v-btn>`
- [ ] "Make the target changable; defaults to window, but can give a valid queryselector or dom element"

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes, `dev` for new features and breaking changes).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have created a PR in the [documentation](https://github.com/vuetifyjs/vuetifyjs.com) with the necessary changes.

### CODE REVIEW

**Comment by jacekkarczmarczyk** [comment]
I don't think `dev/ToTop.vue` file should be included in the PR

File: src/components/Vuetify/mixins/goTo.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+/**
+ * Modified from https://github.com/alamcordeiro/vue-smooth-scroll
+ */
+
+function easeInOutCubic (t) { return t < .5 ? 4 * t * t * t : (t - 1) * (2 * t - 2) * (2 * t - 2) + 1 }
+
+export default function (target) {
```
Comment (KaelWD) [inline src/components/Vuetify/mixins/goTo.js]: Give this function a name.

File: src/components/Vuetify/mixins/goTo.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  if (typeof window === 'undefined') return
+
+  let duration = 500
+  let start = Date.now()
```
Comment (KaelWD) [inline src/components/Vuetify/mixins/goTo.js]: `Date.now()` is not suitable for animation, use `performance.now()` instead.

https://developers.google.com/web/tools/lighthouse/audits/date-now

File: src/components/Vuetify/mixins/goTo.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+  function step () {
+    let elapsed = Date.now() - start
```
Comment (KaelWD) [inline src/components/Vuetify/mixins/goTo.js]: And here you should use the timestamp that is passed by `requestAnimationFrame`

https://developer.mozilla.org/en-US/docs/Web/API/window/requestAnimationFrame

File: src/components/Vuetify/mixins/goTo.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    let position = target
+    if (elapsed < duration){
+      position = window.pageYOffset + (target - window.pageYOffset) * easeInOutCubic(elapsed / duration)
+      window.requestAnimationFrame(step)
+    }
+    window.scroll(0, position)
+  }
+
+  step()
```
Comment (KaelWD) [inline src/components/Vuetify/mixins/goTo.js]: This should be in a `requestAnimationFrame` too.

**Pull request closed without merging**